lib/sqlalchemy/ext/asyncio/session.py

```diff
@@ -1436,6 +1436,40 @@ async def main():
 
     class_: Type[_AS]
 
+    @overload
+    def __init__(
+        self: async_sessionmaker[AsyncSession],
```

Comment:
This overload does not have `class_`, so mypy will then infer the type as `[AsyncSession]` from the self argument, when no class_ is provided.